test/tools/cmap_spec_runner.ts

```diff
@@ -372,6 +390,102 @@ export async function runCmapTest(test: CmapTest, threadContext: ThreadContext)
     const actual = actualEvents.shift();
     const { type: eventType, ...eventPropsToCheck } = expected;
     expect(actual.constructor.name).to.equal(`${eventType}Event`);
-    compareInputToSpec(actual, eventPropsToCheck);
+    compareInputToSpec(actual, eventPropsToCheck, `failed while checking ${eventType} event`);
+  }
+}
+
+export type SkipDescription = {
+  description: string;
+  skipIfCondition: 'loadBalanced';
+  skipReason: string;
+};
+
+export function runCmapTestSuite(
+  tests: CmapTest[],
+  options?: { testsToSkip?: SkipDescription[]; injectPoolStats?: boolean }
+) {
+  for (const test of tests) {
+    describe(test.name, function () {
+      let hostAddress: HostAddress, threadContext: ThreadContext, client: MongoClient;
+
+      beforeEach(async function () {
+        let utilClient: MongoClient;
+
+        const skipDescription = options?.testsToSkip?.find(
+          ({ description }) => description === test.description
+        );
+        if (skipDescription) {
+          const matchesLoadBalanceSkip =
+            skipDescription.skipIfCondition === 'loadBalanced' && this.configuration.isLoadBalanced;
+          if (matchesLoadBalanceSkip) {
+            (this.currentTest as Mocha.Test & { skipReason?: string }).skipReason =
```

Comment:
If you open the global.d.ts file you should be able to get the correct type info, not ideal but I'd rather not add casting. Or we can mark it with `TODO(NODE-4247)`

Reply:
are you asking me to add `currentTest` to that list with our `Mocha.Runnable` type? because if not, it would still be a cast here

Reply:
`currentTest` is typed as `Mocha.Test` which extends `Mocha.Runnable` which is where we define the `skipReason`, so it should be available on this variable. Is it giving you type error anyway?

Reply:
I wouldn't have updated this line if `this.currentTest.skipReason =` didn't yield a red squiggly

Reply:
(Not sure where you're seeing the extends Runnable definitions here, but they're not getting picked up)